### PR TITLE
[SPDBT-3005] added optionsets and enums PoliceOfficerRole

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):

- PoliceOfficerRoleEnum
- PoliceOfficerRoleCode
- PoliceOfficerRoleOptionSet

### CODE REVIEW

File: src/Spd.Manager.Licence/SharedContract.cs [hunk 1]
```diff
@@ -89,7 +89,9 @@ public enum PoliceOfficerRoleCode
     CourtAppointedBailiff,
     SpecialProvincialOrMunicipalConstable,
     PoliceOfficer,
+    PoliceOfficerRetired,
     Other,
+    None,
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/SharedContract.cs:94]: 'None' should not be visible to the frontend. Please remove.

Comment (esdd1995) [reply src/Spd.Manager.Licence/SharedContract.cs:94]: removed

File: src/Spd.Manager.Licence/SharedContract.cs [hunk 2]
```diff
@@ -89,6 +89,7 @@ public enum PoliceOfficerRoleCode
     CourtAppointedBailiff,
     SpecialProvincialOrMunicipalConstable,
     PoliceOfficer,
+    PoliceOfficerRetired,
     Other,
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/SharedContract.cs:93]: do we need to add code in read, if it returned None, then we return null?

Comment (peggy-quartech) [reply src/Spd.Manager.Licence/SharedContract.cs:93]: the same for create, if user input null, should we set it to None in dynamics?

Comment (esdd1995) [reply src/Spd.Manager.Licence/SharedContract.cs:93]: I removed None form both `PoliceOfficerRoleCode` and `PoliceOfficerRoleEnum` enums.
just have None in optionset, which in the setting and getting function I've handled it

*fe sends and receives null, however the source is `null` or `None`
@carolcarpenter

Comment (peggy-quartech) [reply src/Spd.Manager.Licence/SharedContract.cs:93]: please also deal with the oppsite direction, if fe set to null, then be needs to set None in dynamics.

Comment (esdd1995) [reply src/Spd.Manager.Licence/SharedContract.cs:93]: it is doing this way now, just save None, never saves or updates to null in db

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**